docs/proposals/20230726-integrate-keda-http-add-on.md

```diff
@@ -0,0 +1,285 @@
+# Integrate KEDA http-add-on
+
+## Motivation
+OpenFunction has released v1.1.1, and [keda-http-addon](https://github.com/kedacore/http-add-on/) has been updated to v0.5.0. Now we can try to add support for keda-http-addon.
+
+In OpenFunction v1beta2 version API, `trigger` and `inputs` have been separated, so keda-http-addon can take over the part of `HttpTrigger`; on the other hand, integrating keda-http-addon also helps to unify the core architecture of OpenFunction and improve consistency. 
```

Comment:
We might need to add something like `engine` or `type` into `HttpTrigger`, can be set to `knative` or `keda`, default to Knative if not set

https://github.com/OpenFunction/OpenFunction/blob/main/apis/core/v1beta2/serving_types.go#L46